File: solutions/javascript/bottle-song/1/bottle-song.js
// For my Shree DR.MDD

export const recite = (startingCount, removeCount) => {
  let lyrics = [];
  for (let current = startingCount; current > (startingCount - removeCount); current--) {
    lyrics.push(...singVerse(current));
    if(removeCount > 1 && current - 1 !== (startingCount - removeCount)) lyrics.push('');
  }
  return lyrics;
};

function singVerse(count) {
  let countWords = [
    "no",
    "One",
    "Two",
    "Three",
    "Four",
    "Five",
    "Six",
    "Seven",
    "Eight",
    "Nine",
    "Ten",
  ];
  let line1 = `${countWords[count]} green bottle${count !== 1 ? "s" : ""} hanging on the wall,`;
  let line2 = `And if one green bottle should accidentally fall,`;
  let line3 = `There'll be ${countWords[count - 1].toLowerCase()} green bottle${count - 1 !== 1 ? "s" : ""} hanging on the wall.`;
  return [line1, line1, line2, line3];
}

File: solutions/go/bottle-song/1/bottle_song.go
// Shree DR.MDD
package bottlesong

import (
	"fmt"
	"strings"
)

var numWords = []string{"No", "One", "Two", "Three", "Four", "Five",
	"Six", "Seven", "Eight", "Nine", "Ten"}

func Recite(start, count int) []string {
	var verses []string
	stop := start - count
	for i := start; i > stop; i-- {
		if i < start {
			verses = append(verses, "")
		}
		verses = append(verses, makeVerse(i)...)
	}
	return verses
}

func makeVerse(n int) []string {
	line1 := fmt.Sprintf("%s green bottle%s hanging on the wall,", numWords[n], addS(n))
	line2 := "And if one green bottle should accidentally fall,"
	line3 := fmt.Sprintf("There'll be %s green bottle%s hanging on the wall.", strings.ToLower(numWords[n-1]), addS(n-1))
	return []string{line1, line1, line2, line3}
}

func addS(n int) string {
	if n != 1 {
		return "s"
	}
	return ""
}
